src/prefect/engine.py

```diff
@@ -2158,14 +2158,16 @@ async def _run_task_hooks(task: Task, task_run: TaskRun, state: State) -> None:
 
 
 async def _run_flow_hooks(flow: Flow, flow_run: FlowRun, state: State) -> None:
-    """Run the on_failure and on_completion hooks for a flow, making sure to
+    """Run the on_failure, on_completion, and on_cancellation hooks for a flow, making sure to
```

Comment:
Flows usually will encounter cancellation via a SIGTERM which I presume bypasses this?

Reply:
After some investigation, it seems to bypass it sometimes. Do you think it would make sense to move calling the cancellation hook to where we check for cancelled flow runs (flow runs in a `CANCELLING` state)?